fbpic/boundaries/boundary_communicator.py

```diff
@@ -208,16 +256,32 @@ def divide_into_domain( self, zmin, zmax, p_zmin, p_zmax ):
                                than 4 times n_guard. Use fewer domains or \
                                a smaller number of guard cells.')
 
-        # Calculate the local boundaries (zmin and zmax)
+        # Calculate the local boundaries (zmin)
         # of this local simulation box including the guard cells.
         iz_start = self.iz_start_procs[self.rank]
         zmin_local = zmin + (iz_start - self.n_guard)*dz
```

Comment:
This whole section of code could probably be simplified by doing:
```python
zmin_local_domain = zmin + iz_start*dz
zmax_local_domain = zmin_local_domain + self.Nz_domain
p_zmin = max( zmin_local_domain + self.n_guard*dz, p_zmin)
p_zmax = min( zmax_local_domain - self.n_guard*dz, p_zmax)
```
where `zmin_local_domain` and `zmax_local_domain` are the physical positions of the local **physical domain** (i.e. without guard and damp cells).